Lib/defcon/objects/contour.py

```diff
@@ -673,6 +685,93 @@ def endSelfNotificationObservation(self):
         self._layer = None
         self._glyph = None
 
+    # -----------------------------
+    # Serialization/Deserialization
+    # -----------------------------
+
+    def getDataForSerialization(self, **kwargs):
+        def get_points(key):
+            # store the point pen protocol calls
+            # this will store the identifier and the point data
+            pointData = []
+            self.drawPoints(Recorder(pointData))
+            return pointData
+
+        getters = [('pen', get_points)]
+        return self._serialize(getters, **kwargs)
+
+    def setDataFromSerialization(self, data):
+        self.clear()
+        self.identifier = None
+        if 'pen' in data:
+            # play back
+            Recorder(data['pen'])._play(self)
+
+
+class Recorder(object):
+    """
+        Records all method calls it receives in a list of tuples in the form of
+        [(:str:command, :list:args, :dict: kwargd)]
+
+        Method calls to be recorded must not start with an underscore.
+
+        This class defines a public method with a private(!) attribute name:
+        "Recorder._play(self, target)" because that way calls to all methods
+        that don't start with underscores can be recorded.
```

Comment:
this `Recorder` is really cool. I just don't like that it's only public method `_play` has an underscore...

Wouldn't it be better if we use `__call__` instead of `_play`? The Recorder pen instance would then behave like a callable function.